File: WindowsApplication1/Modules/SQLOperations.vb
Imports System.Data.SqlClient

Public Class SQLOperations

    Public Function GetDataFromDatabase(Query As String) As DataTable
        Dim DT As New DataTable()
        Try
            Call Class1.CONNIC()
            Dim DA As New SqlDataAdapter(Query, cn)
            Dim DS As New DataSet()

            DA.Fill(DS, "MyTable")

            If DS.Tables(0).Rows.Count > 0 Then
                DT = DS.Tables(0)
            Else
                DT = Nothing
            End If
        Catch ex As Exception
            MsgBox($"حدث خطأ أثناء عملية جلب البيانات {ex.Message}", MsgBoxStyle.Information)
        End Try

        Return DT
    End Function

    Public Function GetValueFromDB(Aggregate As String, FieldName As String, TableName As String, Extra As String) As Decimal
        Dim RV As Decimal = 0
        Try
            Call Class1.CONNIC()
            Dim DT As New DataTable()
            Dim DA As New SqlDataAdapter($"Select {Aggregate}({FieldName}) from {TableName} {Extra}", cn)
            Dim DS As New DataSet()

            DA.Fill(DS, "MyTable")
            DT = DS.Tables(0)

            If DT.Rows(0)(0) = DBNull.Value Then
                RV = 0
            Else
                RV = DT.Rows(0)(0)
            End If
        Catch ex As Exception
            MsgBox($"حدث خطأ أثناء عملية جلب البيانات: {ex.Message}", MsgBoxStyle.Information)
        End Try
        Return RV
    End Function
End Class

File: WindowsApplication1/Modules/SQLOperations.cs
using System;
using System.Data;
using System.Data.SqlClient;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace MCS
{

    public class SQLOperations
    {

        public DataTable GetDataFromDatabase(string Query)
        {
            var DT = new DataTable();
            try
            {
                MCS.Class1.CONNIC();
                var DA = new SqlDataAdapter(Query, MCS.Module1.cn);
                var DS = new DataSet();

                DA.Fill(DS, "MyTable");

                if (DS.Tables[0].Rows.Count > 0)
                {
                    DT = DS.Tables[0];
                }
                else
                {
                    DT = null;
                }
            }
            catch (Exception ex)
            {
                Interaction.MsgBox($"حدث خطأ أثناء عملية جلب البيانات {ex.Message}", MsgBoxStyle.Information);
            }

            return DT;
        }

        public decimal GetValueFromDB(string Aggregate, string FieldName, string TableName, string Extra)
        {
            decimal RV = 0m;
            try
            {
                MCS.Class1.CONNIC();
                var DT = new DataTable();
                var DA = new SqlDataAdapter($"Select {Aggregate}({FieldName}) from {TableName} {Extra}", MCS.Module1.cn);
                var DS = new DataSet();

                DA.Fill(DS, "MyTable");
                DT = DS.Tables[0];

                if (DT.Rows.Count > 0)
                {
                    RV = Conversions.ToDecimal(DT.Rows[0][0]);
                }
                else
                {
                    RV = 0m;
                }
            }
            catch (Exception ex)
            {
                // MsgBox($"حدث خطأ أثناء عملية جلب البيانات: {ex.Message}", MsgBoxStyle.Information)
            }
            return RV;
        }
    }
}